File: GXpert/GXpert.Web/Modules/Activation/CouponCode/CouponCodeColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Activation.Columns;

[ColumnsScript("Activation.CouponCode")]
[BasedOnRow(typeof(CouponCodeRow), CheckNames = true)]
public class CouponCodeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    public string PlayListTitle { get; set; }
    [QuickFilter]
    public short ValidityType { get; set; }
    [QuickFilter]
    public short CountType { get; set; }
    public short Count { get; set; }
    public short ValidityInDays { get; set; }
    public DateTime ValidDate { get; set; }
    public int ConsumedCount { get; set; }
    public DateTime CouponValidityDate { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Activation/CouponCodeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EValidityType } from "../Web/Enums.EValidityType";
import { CouponCodeRow } from "./CouponCodeRow";

export interface CouponCodeColumns {
    Id: Column<CouponCodeRow>;
    Code: Column<CouponCodeRow>;
    PlayListTitle: Column<CouponCodeRow>;
    ValidityType: Column<CouponCodeRow>;
    CountType: Column<CouponCodeRow>;
    Count: Column<CouponCodeRow>;
    ValidityInDays: Column<CouponCodeRow>;
    ValidDate: Column<CouponCodeRow>;
    ConsumedCount: Column<CouponCodeRow>;
    CouponValidityDate: Column<CouponCodeRow>;
}

export class CouponCodeColumns extends ColumnsBase<CouponCodeRow> {
    static readonly columnsKey = 'Activation.CouponCode';
    static readonly Fields = fieldsProxy<CouponCodeColumns>();
}

[EValidityType]; // referenced types